src/bucket/BucketController.ts

```diff
@@ -153,6 +181,60 @@ export class BucketController {
       this.allowDownload = true;
     }
 
+    // default sts usage to false
+    if (this.stsApiUrl === undefined) {
+      this.stsEnabled = false;
+    } else {
+      this.stsEnabled = true;
+    }
+
+    // collect sts credentials if needed
+    if (this.stsEnabled) {
+      var httpHeaders;
+      if (this.stsHeaderName === undefined || this.stsHeaderValue === undefined) {
+        httpHeaders = {
+          headers: {
+            'Content-Type': 'application/json'
+          }
+        };
+      } else {
+        var name = this.stsHeaderName;
+        httpHeaders = {
+          headers: {
+            'Content-Type': 'application/json',
+            name: this.stsHeaderValue
+          }
+        };
+      }
+
+      var result: ng.IPromise<any> = this.httpService.post(
```

Comment:
Don't use `any` unless you absolutely cannot avoid it. You could create a type that describes the response and use that instead of `any` here.